src/Config/TerminusConfig.php

```diff
@@ -0,0 +1,194 @@
+     * @return array
+     */
+    public function keys() {
+        return array_keys($this->config);
+    }
+
+    /**
+     * Override the values in this Config object with the given input Config
+     *
+     * @param \Pantheon\Terminus\Config\TerminusConfig $in
+     */
+    public function extend(TerminusConfig $in) {
+        foreach ($in->keys() as $key) {
+            $this->set($key, $in->get($key));
+            // Set the source of this variable to make tracking config easier.
+            $this->setSource($key, $in->getSource($key));
+        }
+    }
+
+    /**
+     * Get the name of the source for this configuration object.
+     *
+     * @return string
+     */
+    public function getSourceName() {
+        return 'Unknown';
+    }
+
+    /**
+     * Get a description of where this configuration came from.
+     *
+     * @param $key
+     * @return string
+     */
+    public function getSource($key) {
+        return isset($this->sources[$key]) ? $this->sources[$key] : $this->getSourceName();
+    }
+
+    /**
+     * Set the source for a given configuration item.
+     *
+     * @param $key
+     * @param $source
+     */
+    protected function setSource($key, $source) {
+        $this->sources[$key] = $source;
+    }
+
+    /**
+     * Reflects a key name given a Terminus constant name
+     *
+     * @param string $constant_name The name of a constant to get a key for
+     * @return string
+     */
+    protected function getKeyFromConstant($constant_name)
+    {
+        $key = strtolower(str_replace($this->constant_prefix, '', $constant_name));
+        return $key;
+    }
+
+    /**
+     * Turn an internal key into a constant name
+     *
+     * @param string $key The key to get the constant name for.
+     * @return string
+     */
+    public function getConstantFromKey($key)
+    {
+        $key = strtoupper($this->constant_prefix . $key);
+        return $key;
+    }
+
+    /**
+     * Determines if a key is a Terminus constant name
+     *
+     * @param $key
+     * @return boolean
+     */
+    protected function keyIsConstant($key) {
+        return strpos($key, $this->constant_prefix) === 0;
+    }
+
+    /**
+     * Exchanges values in [[ ]] in the given string with constants
+     *
+     * @param string $string The string to perform replacements on
+     * @return string $string The modified string
+     */
+    protected function replacePlaceholders($string)
+    {
+        $regex = '~\[\[(.*?)\]\]~';
+        preg_match_all($regex, $string, $matches);
+        if (!empty($matches)) {
+            foreach ($matches[1] as $id => $value) {
+                $replacement_key = $this->getKeyFromConstant(trim($value));
+                $replacement = $this->get($replacement_key);
+                if ($replacement) {
+                    $string = str_replace($matches[0][$id], $replacement, $string);
+                }
+            }
+        }
+        $fixed_string = $this->fixDirectorySeparators($string);
```

Comment:
Is this actually necessary? AFAIK php allows either direction of slash on Window (so that we can avoid this kind of thing). I don't have a working Windows setup though so I'm not sure about that.